pyop2/mpi.py

```diff
@@ -465,26 +474,41 @@ def compilation_comm(comm):
     else:
         comp_comm = comm
     incref(comp_comm)
+    weakref.finalize(obj, decref, comp_comm)
     return comp_comm
 
 
+def finalize_safe_debug():
+    if PYOP2_FINALIZED:
+        if logger.level > DEBUG and not _running_on_ci:
+            debug = lambda string: None
+        else:
+            debug = lambda string: print(string)
+    return debug
+
+
 @atexit.register
 def _free_comms():
     """Free all outstanding communicators."""
     global PYOP2_FINALIZED
     PYOP2_FINALIZED = True
-    if logger.level > DEBUG:
-        debug = lambda string: None
-    else:
-        debug = lambda string: print(string)
+    debug = finalize_safe_debug()
     debug("PyOP2 Finalizing")
     # Collect garbage as it may hold on to communicator references
+
     debug("Calling gc.collect()")
     gc.collect()
+    debug("STATE0")
```

Comment:
Not a very helpful statement to have here.

Reply:
Do you have a suggestion, I was only using `STATE {0,1,2}` for debugging and was potentially going to remove the "titles"